tools/validators/instance_validator/README.md

```diff
@@ -41,15 +41,16 @@ Note: as of the current development stage, you must clone the entire repository
 
 The validator supports a telemetry validation mode. When this mode is enabled, the validator will listen on a provided pubsub subscription for telemetry messages, and validate the message contents against the instance configuration. It is recommended that you first use the instance validator with telemetry validation mode disabled, and then enable it after that passes.
 
-If you would like to use the telemetry validation mode, you must provide the "--subscription" and "--service-account" parameters when running instance_validator.py. Failure to provide both of these parameters will result in early termination of the validator and an error message. If you do not provide either parameter, the validator will run with telemetry validation mode disabled.
+If you would like to use the telemetry validation mode, you must provide the "--subscription", and you must either be authentication with gcloud application-default credentials, or provide a "--service-account" parameters when running instance_validator.py. Failure to provide  these parameters will result in early termination of the validator and an error message. If you do not provide either parameter, the validator will run with telemetry validation mode disabled.
```

Comment:
can you please add something like `with the adequate google project id provided by your google point of contact` or something in those lines?

Reply:
Interesting, didn't capture this in testing but I think I need to add an additional parameter for the project , because normal user accounts can't set the google projects as the default project. So either the full subscription path `projects/<PROJECT_ID>/subscriptions/<SUBSCRIPTION_NAME>`needs to be given or an additional parameter `project_id`. Will update the PR to incorporate this for review

Reply:
Actually I understand current guidance is putting the full subscription path so no further changes needed

Reply:
[sorry for the delay, just seeing this, please trigger the review when you are ready :) ]

correct but the user must be have the correct id with the gcloud auth otherwise they will get an error of unauthentication. what do you think?

Reply:
Sorry just picking this up again. I don't think it's needed for actually, I've tested on my machine by removing gcloud credentials and settings, and only setting up the application-default credentials and it worked for me. I want to test on a fresh install to confirm though, so will post back.